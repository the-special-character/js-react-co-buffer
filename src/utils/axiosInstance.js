import axios from 'axios';

const axiosInstance = axios.create({
  baseURL: 'http://localhost:3000',
  timeout: 5000,
  timeoutErrorMessage: 'Try after some time',
});

export default axiosInstance;
