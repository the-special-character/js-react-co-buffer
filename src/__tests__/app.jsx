import React from 'react';
import {
  render,
  screen,
  waitFor,
  fireEvent,
  waitForElementToBeRemoved,
} from '@testing-library/react';
import MockAdapter from 'axios-mock-adapter';
import App from '../app';
import axiosInstance from '../utils/axiosInstance';

var mock = new MockAdapter(axiosInstance);

const todoList = [
  {
    text: 'hello',
    isDone: false,
    id: 9,
  },
];

describe('App Component', () => {
  test('should render app component', async () => {
    mock.onGet('/todoList').reply(200, todoList);
    const { asFragment } = render(<App />);
    await waitForElementToBeRemoved(() =>
      screen.queryByText(/Loading.../i),
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test('should render loading before api call', async () => {
    mock.onGet('/todoList').reply(200, todoList);
    render(<App />);
    await waitForElementToBeRemoved(() =>
      screen.queryByText(/Loading.../i),
    );
  });

  test('should render todo app', async () => {
    mock.onGet('/todoList').reply(200, todoList);
    render(<App />);
    const title = await screen.findByText('Todo App');
    expect(title).toBeInTheDocument();
  });

  test('should complete add functionality', async () => {
    mock.onGet('/todoList').reply(200, todoList);
    mock.onPost('/todoList').reply(200, {
      text: 'hello world',
      isDone: false,
      id: 5,
    });
    render(<App />);
    await waitForElementToBeRemoved(() =>
      screen.queryByText(/Loading.../i),
    );
    const todoInput = screen.queryByLabelText(/Todo Text/i);
    fireEvent.change(todoInput, {
      target: { value: 'hello world' },
    });
    expect(todoInput.value).toBe('hello world');

    const addTodo = screen.queryByText(/Add Todo/i);
    fireEvent.click(addTodo);
    await waitFor(() =>
      expect(
        screen.queryAllByTestId('todolist-item').length,
      ).toBe(2),
    );
  });
});
