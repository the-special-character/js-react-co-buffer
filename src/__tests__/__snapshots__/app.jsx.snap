// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component should render app component 1`] = `
<DocumentFragment>
  <div
    class="todo"
  >
    <h1
      class="todo__title"
    >
      Todo App
    </h1>
    <form
      class="todo__form"
    >
      <div>
        <label
          class="sr-only"
          for="todoText"
        >
          Todo Text
        </label>
        <input
          class="rounded-l-md"
          id="todoText"
          type="text"
        />
      </div>
      <button
        class="btn rounded-r-md disabled:bg-slate-500 disabled:cursor-wait"
        type="submit"
      >
        Add Todo
      </button>
    </form>
    <div
      class="todo__list"
    >
      <div
        class="todo__list-item"
        data-testid="todolist-item"
      >
        <div>
          <label
            class="sr-only"
            for="isDone"
          >
            Completed
          </label>
          <input
            class="disabled:text-slate-400 disabled:cursor-wait"
            id="isDone"
            type="checkbox"
          />
        </div>
        <p
          class="px-4 flex-1"
        >
          hello
        </p>
        <button
          class="btn rounded-md disabled:bg-slate-500 disabled:cursor-wait"
          type="button"
        >
          Delete
        </button>
      </div>
    </div>
    <div
      class="todo__filter"
    >
      <button
        class="btn flex-1 btn--active"
        type="button"
      >
        All
      </button>
      <button
        class="btn flex-1"
        type="button"
      >
        Pending
      </button>
      <button
        class="btn flex-1"
        type="button"
      >
        Completed
      </button>
    </div>
  </div>
</DocumentFragment>
`;
